components/scream/src/dynamics/homme/physics_dynamics_remapper.hpp

```diff
@@ -102,13 +101,56 @@ class PhysicsDynamicsRemapper : public AbstractRemapper<RealType>
 public:
   // These functions should be morally privade, but CUDA does not allow extended host-device lambda
   // to have private/protected access within the class
-  void local_remap_fwd_2d (const field_type& phys, const field_type& dyn, const int itl) const;
-  template<typename ScalarT>
-  void local_remap_fwd_3d_impl (const field_type& phys, const field_type& dyn, const int itl) const;
 
-  void remap_bwd_2d (const field_type& phys, const field_type& dyn, const int itl) const;
-  template<typename ScalarT>
-  void remap_bwd_3d_impl (const field_type& phys, const field_type& dyn, const int itl) const;
+  struct Pointer {
+    KOKKOS_FORCEINLINE_FUNCTION
+    Real* get() { return ptr; }
+
+    Real* ptr;
+  };
+
+  KokkosTypes<DefaultDevice>::view_1d<Pointer> phys_ptrs;
+  KokkosTypes<DefaultDevice>::view_2d<Int>     phys_dims;
```

Comment:
I am thinking that, since the 2nd dim is known at compile-time, we might be better off with

```
view_1d<Kokkos::Array<Int,4>> phys_dims;
```
This might save indexing time (is this true?) and also register count when using it (see below comment marked "ABOUT PHYS_DIMS").